go/tasks/plugins/array/awsbatch/definition/job_def_cache.go

```diff
@@ -27,12 +29,13 @@ type CacheKey interface {
 }
 
 type cacheKey struct {
-	role  string
-	image string
+	role               string
+	image              string
+	jobDefinitionInput batch.RegisterJobDefinitionInput
 }
 
 func (k cacheKey) String() string {
-	return fmt.Sprintf("%v-%v", k.image, k.role)
+	return fmt.Sprintf("%v-%v-%v", k.image, k.role, k.jobDefinitionInput.String())
```

Comment:
I think we should hash it? maybe fnv, otherwise this will be large memory overhead.

Reply:
cc @EngHabu do we need the entire definition? It looks like name should be enough [here](https://docs.aws.amazon.com/batch/latest/userguide/create-job-definition.html)

Reply:
but the different tasks have the same job definition name now. The name is the same as the image tag.
If image tag is `pingsutw/flyte-app:b22d5edf6b1777a2bcfeef2465e6a7a6`, the job definition will be `flyte-app`

Reply:
btw, should we replace freecache with bigcache. It seems like bigcache is faster than freecache. check this [benchmark](https://github.com/allegro/bigcache#benchmarks)
If so, I can create PR for this.